src/relentless/model/potential/angle.py

```diff
@@ -0,0 +1,397 @@
+        If ``mode='diff'``, the amplitude of the *last* knot is fixed, and
+        differences between neighboring knots are manipulated for all other
+        knots. Defaults to ``'diff'``.
+    name : str
+        Unique name of the potential. Defaults to ``__u[id]``, where ``id`` is the
+        unique integer ID of the potential.
+
+    Examples
+    --------
+    The spline potential is setup from a tabulated potential instead of
+    specifying knot parameters directly::
+
+        spline = relentless.potential.angle.AngleSpline(types=("angleA",), num_knots=3)
+        spline.from_array(("angleA"),[0,1,2],[4,2,0])
+
+    However, the knot variables can be iterated over and manipulated directly::
+
+        for r,k in spline.knots("angleA"):
+            k.value = 1.0
+
+    """
+
+    _space_coord_name = "theta"
+
+    def __init__(self, types, num_knots, mode="diff", name=None):
+        super().__init__(types=types, num_knots=num_knots, mode=mode, name=name)
+
+    def from_array(self, types, theta, u):
+        r"""Set up the potential from knot points.
+
+        Parameters
+        ----------
+        types : tuple[str]
+            The type for which to set up the potential.
+        theta : list
+            Position of each knot.
+        u : list
+            Potential energy of each knot.
+
+        Raises
+        ------
+        ValueError
+            If the number of ``theta`` values is not the same as the number of knots.
+        ValueError
+            If the number of ``u`` values is not the same as the number of knots.
+
+        """
+
+        if theta[0] != 0.0 and theta[-1] != numpy.pi:
+            raise ValueError("The first and last knot must be at 0 and pi.")
+        return super().from_array(types=types, x=theta, u=u)
+
+    def energy(self, type_, theta):
+        """Evaluate potential energy.
+
+        Parameters
+        ----------
+        type_
+            Type parametrizing the potential in :attr:`coeff<container>`.
+        theta : float or list
+            Potential energy coordinate.
+
+        Returns
+        -------
+        float or numpy.ndarray
+            The pair energy evaluated at ``theta``. The return type is consistent
+            with ``theta``.
+
+        """
+        return super().energy(type_=type_, x=theta)
+
+    def force(self, type_, theta):
+        """Evaluate force magnitude.
+
+        The force is the (negative) magnitude of the ``theta`` gradient.
+
+        Parameters
+        ----------
+        type_
+            Type parametrizing the potential in :attr:`coeff<container>`.
+        theta : float or list
+            Potential energy coordinate.
+
+        Returns
+        -------
+        float or numpy.ndarray
+            The force evaluated at ``theta``. The return type is consistent
+            with ``theta``.
+
+        """
+        return super().force(type_=type_, x=theta)
+
+    def _derivative(self, param, r, **params):
```

Comment:
This should also be `theta`
```suggestion
    def _derivative(self, param, theta, **params):
```
The `derivative` function also needs to have a similar pass-through implementation.